swift/Samples/SplitScreenContainer/Sources/SplitScreenContainerViewController.swift

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Workflow
+import WorkflowUI
+
+public final class SplitScreenContainerViewController<LeftScreenType: Screen, RightScreenType: Screen>: ScreenViewController<SplitScreenContainerScreen<LeftScreenType, RightScreenType>> {
+
+    private var leftContentViewController: ScreenViewController<LeftScreenType>? = nil
+    private lazy var leftContainerView: UIView = .init()
+    private var leftContainerViewWidthConstraint: NSLayoutConstraint?
+
+    private lazy var separatorView: UIView = .init()
+
+    private var rightContentViewController: ScreenViewController<RightScreenType>? = nil
+    private lazy var rightContainerView: UIView = .init()
+
+    required init(screen: SplitScreenContainerScreen<LeftScreenType, RightScreenType>, viewRegistry: ViewRegistry) {
+        super.init(screen: screen, viewRegistry: viewRegistry)
+    }
+
+    override public func screenDidChange(from previousScreen: SplitScreenContainerScreen<LeftScreenType, RightScreenType>) {
+        update(with: screen)
+    }
+
+    private func update(with screen: SplitScreenContainerScreen<LeftScreenType, RightScreenType>) {
+        if (separatorView.backgroundColor != screen.separatorColor) {
+            separatorView.backgroundColor = screen.separatorColor
+        }
+        
+        leftContentViewController?.update(screen: screen.leftScreen)
```

Comment:
I had logic in my original version that allowed the left screen and right screen to change type, but I guess you removed that as an option because the screen now generic on the type of screen. Did you consider cases where the screen types might change? why don't we want to allow for that?

Reply:
We made the `SplitScreenContainerViewController` use generic types for `LeftScreenType` and `RightScreenType`. Your correct they can't be changed from within the `SplitScreenContainerViewController` itself, but if you use `AnyScreen` in the client code it will allow you to change the screen. 

Another way to say this is that we can rely on the capabilities built into `AnyScreen` and `AnyScreenViewController` to do the screen swapping for us.  I'll add an example for this.

https://github.com/square/workflow/pull/925/files#diff-0affef380bcecd01beb16c4a5a06b116R47-R53

Reply:
@bizzemfrog please checkout https://github.com/square/workflow/pull/925/commits/6e454a97b338782df3479ac1e574a092d7523ae8 to see if the demo addresses your concerns.

Reply:
I also updated the attached gif to demonstrate the sample changing the left screen and respecting the ratio changes.